src/api/payments.js

```diff
@@ -0,0 +1,42 @@
+import { httpRequest } from "../request/index.js";
+
+const getBaseUrl = () => {
+  const { BACKEND_URL } = window[Symbol.for("app-config")];
+
+  return new URL("payments/", BACKEND_URL);
+};
+
+export const getClientSecret = async (data = []) => {
+  const result = await httpRequest.post(getBaseUrl(), {
+    body: JSON.stringify({
+      products: data.map((item) => {
+        item.quantity = item.count;
+        delete item.count;
```

Comment:
**Issue**: Direct manipulation and deletion of object properties can lead to unexpected side effects, especially when dealing with objects that might be used elsewhere. <br> **Fix**: Use a more functional approach by creating a new object with the required properties instead of deleting properties from the existing object. <br> **Code Suggestion**: 
 ```
 - item.quantity = item.count;
 - delete item.count;
 - delete item.inStore;
 - delete item.id;
 + const newItem = {
 +   quantity: item.count,
 +   ...item,
 + };
 + delete newItem.count;
 + delete newItem.inStore;
 + delete newItem.id;
 + return newItem;
 ```